core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -502,12 +507,16 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
                              producerEpoch: Short,
                              txnMarkerResult: TransactionResult,
                              isFromClient: Boolean,
+                             clientTransactionVersion: Short,
```

Comment:
At first, I was a bit puzzle by this version here because it was not clear to me whether it was the version of the RPC or the TV. In order to make it explicit, I wonder if we should pass the `TransactionVersion` here. If possible, I would also advice to `clientTransactionVersion >= 2` condition within a method in `TransactionVersion`. We did this in the `GroupVersion` and I find it pretty elegant. What do you think?

Reply:
The one thing that is tricky here is we also need to encode that the client sent the newest request (or in other words -- the client + the server support the new version and the client acknowledges that)

Reply:
We discussed offline and I realized he was talkinga bout the type. I have also added the specified method.